compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -772,6 +772,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         let msg = "introduce a new binding instead";
                         let sugg = format!("other_{}", ident.as_str().to_lowercase());
                         e.span_suggestion(ident.span, msg, sugg, Applicability::HasPlaceholders);
+                        match res {
+                            Res::Def(DefKind::Const, _) => {
```

Comment:
This should ideally check for all range types whose syntax can be used in patterns and suggest something more specific. The user likely wanted to match on a range of values instead.

Reply:
ah thats a good idea. Is there an example that might show how to check something like that?

Reply:
Not sure off the top of my head. You can probably get help on Zulip though.